### PR TITLE
Add struct to structpb method that ignores omit empty tags

### PR DESCRIPTION
**Pull request by vijayvuyyuru**
Hey yall, as part of queryable sensor data , we would like users to be able to see all fields of their data regardless of it is an "empty value". If a boolean is false, it can be dropped from the collector which is not ideal. The testing for the new method is fairly barebones as its fundamentally the same method, just added a case where it shouldn't be dropping a variable. I'm happy to add more if required.

### CODE REVIEW

File: protoutils/protoutils_test.go
Comment (agreenb) [inline protoutils/protoutils_test.go:333]: Can you add the same test to TestStructToStructPb and ensure that it instead equals {} there?

File: protoutils/protoutils.go
Comment (agreenb) [inline protoutils/protoutils.go:182]: [nit] shouldOmit -> shouldOmitEmpty, and `tagged omit empty` -> `tagged omitempty`

File: protoutils/protoutils.go
Comment (agreenb) [inline protoutils/protoutils.go:51]: To not duplicate the inner logic, could you have a interfaceToMap helper which takes `data` and `shouldOmitEmpty`? InterfaceToMap can call that with `true`, and then you can get rid of `interfaceToMapKeepEmpty` and just call the helper directly from `StructToStructPbIgnoreOmitEmpty`.

File: protoutils/protoutils_test.go
Comment (agreenb) [inline protoutils/protoutils_test.go:184]: Can extract `true` into a `shouldOmitEmpty` const that's used throughout

**Review by agreenb — CHANGES_REQUESTED** [review]
Nice work on updating the goutils code to work for our desired capture/sync behavior! Just a few things, but overall looks good.

File: protoutils/protoutils_test.go
Comment (agreenb) [inline protoutils/protoutils_test.go:382]: Ah I missed this before - you can revert and I see that there's a test for `{"omit struct", OmitStruct{}, map[string]interface{}{"x": 0.0}`, thanks!

File: protoutils/protoutils.go
Comment (dmhilly) [inline protoutils/protoutils.go:72]: [minor] You can use `errors.Wrapf` here to remove the need for fmt.Sprintf.

File: protoutils/protoutils_test.go
Comment (dmhilly) [inline protoutils/protoutils_test.go:338]: I think I'm missing something; where is OmitStruct defined?

File: protoutils/protoutils_test.go
Comment (dmhilly) [inline protoutils/protoutils_test.go:337]: [nit] Just an FYI that it's okay to not run this as a subtest (within t.Run) since you only have logic for one test in this function. t.Run just lets you differentiate between different subtests within a given Test function (and do useful things like run them in parallel, in isolation, etc.). If you did this in anticipation of adding future tests or to have a more descriptive subtest name then it's okay to keep it the way you have it here, just wanted you to be aware!

**Review by dmhilly — CHANGES_REQUESTED** [review]
Nice work Vijay! Left a couple comments.

**Review by AaronCasas — CHANGES_REQUESTED** [review]
Just one small comment on the `shouldPmitEmpty` naming, and +1 on Devin's question about OmitEmpty. Overall looks great

File: protoutils/protoutils.go
Comment (vijayvuyyuru) [reply protoutils/protoutils.go:19]: Yes you are correct. I'll change the name

File: protoutils/protoutils_test.go
Comment (vijayvuyyuru) [reply protoutils/protoutils_test.go:337]: Ah makes sense, ill change it.

File: protoutils/protoutils_test.go
Comment (vijayvuyyuru) [reply protoutils/protoutils_test.go:338]: https://github.com/viamrobotics/goutils/blob/c6d1b7e6309fff9e05367c8c0f0628a6dcb74c88/protoutils/protoutils_test.go#L372 Its at the bottom of the file. It just has a field that is tagged as omitEmpty.

**Review by AaronCasas — APPROVED** [review]
LGTM once tests are passing

**Review by agreenb — APPROVED** [review]
Really nice work!

File: protoutils/protoutils.go
Comment (dmhilly) [inline protoutils/protoutils.go:16]: I'm confused, now it looks like there is a mixture of ignoreOmitEmpty and shouldOmitEmpty. If we're going with ignoreOmitEmpty, then this should be false, correct?

File: protoutils/protoutils.go
Comment (dmhilly) [inline protoutils/protoutils.go:69]: Should this be `true`?

File: protoutils/protoutils_test.go
Comment (dmhilly) [reply protoutils/protoutils_test.go:335]: Spoke with Vijay offline; he didn't end up adding a test for this because one already exists [here](https://github.com/viamrobotics/goutils/pull/204/files#diff-9c3c3101f109f0429d739b33ceb0acd4463bad9e25bc8c8d1b2769d000a7c501R75)

Alexa had commented this elsewhere but I missed it

**Review by agreenb — COMMENTED** [review]
Request to run all RDK tests locally while referencing this goutils version before merging, to ensure that all still passes. Nice work Vijay!

**Review by dmhilly — APPROVED** [review]
LGTM! Agree with Alexa about running tests locally. Nice work 😄

**Pull request merged**
